src/interface/web/app/settings/page.tsx

```diff
@@ -0,0 +1,1018 @@
+                                                ?
+                                                    <Button variant="outline" size="sm" onClick={() => syncContent("notion")}>
+                                                        <ArrowsClockwise className="h-5 w-5 inline mr-1" />Sync
+                                                    </Button>
+                                                /* Show set API key button notion oauth url not set setup */
+                                                : !userConfig.notion_oauth_url
+                                                ?
+                                                    <Button variant="outline" size="sm" onClick={saveNotionToken} disabled={notionToken === userConfig.notion_token}>
+                                                        <FloppyDisk className="h-5 w-5 inline mr-1" />
+                                                        {userConfig.enabled_content_source.notion && "Update API Key" || "Set API Key"}
+                                                    </Button>
+                                                : <></>
+                                                )}
+                                            <Button
+                                                variant="outline"
+                                                size="sm"
+                                                className={`${userConfig.notion_token || "hidden"}`}
+                                                onClick={() => disconnectContent("notion")}
+                                            >
+                                                <CloudSlash className="h-5 w-5 inline mr-1" />Disconnect
+                                            </Button>
+                                        </CardFooter>
+                                    </Card>
+                                </div>
+                            </div>
+                            <div className="section grid gap-8">
+                                <div className="text-2xl">Models</div>
+                                <div className="cards flex flex-wrap gap-16">
+                                    {userConfig.chat_model_options.length > 0 && (
+                                    <Card className={cardClassName}>
+                                        <CardHeader className="text-xl flex flex-row"><ChatCircleText className="h-7 w-7 mr-2"/>Chat</CardHeader>
+                                        <CardContent className="overflow-hidden pb-12 grid gap-8">
+                                            <p className="text-gray-400">Pick the chat model to generate text responses</p>
+                                            <DropdownComponent
+                                                items={userConfig.chat_model_options}
+                                                selected={userConfig.selected_chat_model_config}
```

Comment:
Should disable the callback if the user is not subscribed. Same for the other cards.